src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala

```diff
@@ -2083,23 +2146,35 @@ private def checkStatement(
     }
 
     private def checkContract(contract: Contract): Contract = {
-        currentContractSourcePath = contract.sourcePath
 
-        val table = globalTable.contractLookup(contract.name)
-        val newDecls = contract.declarations.map(checkDeclaration(table))
+        contract match {
+            case obsContract : ObsidianContractImpl =>
+                val table = globalTable.contractLookup(obsContract.name)
+                val newDecls = obsContract.declarations.map(checkDeclaration(table))
 
-        checkConstructors(table.constructors, contract, table)
+                checkConstructors(table.constructors, obsContract, table)
 
-        contract.copy(declarations = newDecls)
+                obsContract.copy(declarations = newDecls)
+            case ffiContract : javaFFIContractImpl => ffiContract
+        }
     }
 
+    //private def checkFFIContract(contract : javaFFIContractImpl): Unit = {
```

Comment:
Remove this, or call it but put a TODO comment in the body to check to make sure the Java class matches the Obsidian specification.